### ISSUE TITLE
Support nullable types in DB schema

### ISSUE THREAD

**Issue by Dodje** [ardanlabs/service#410]
Hi!
It might be a topic to discuss, but I'd want to propose to use `pgtype` for DB layer struct when we  do "select queries" and others from postgres.
As I know lots of DBA's don't tolerate empty string inside columns when there nulls are expected. That's why I've started to create schema with NOT NULL props and found that it's not possible to caste "AnyColumn string `db:"any_column"`" with the error 
`converting NULL to string is unsupported`.

So I suggest to use `pgtype.Text` by default wich allows us to use nullable conversion by default.

**Comment by ardan-bkennedy** [comment]
Where is the definition of this type `pgtype.Text`? Are you are talking about this package?

https://github.com/jackc/pgtype

If you are suggesting to bring this package in just for a single type and not for any of its behavior. I'm not a fan of that.

We have a NULL type in the standard library `sql.Null` that I do use when NULL is a valid option for a field.

```
type user struct {
	ID           uuid.UUID      `db:"user_id"`
	Name         string         `db:"name"`
	Email        string         `db:"email"`
	Roles        dbarray.String `db:"roles"`
	PasswordHash []byte         `db:"password_hash"`
	Department   sql.NullString `db:"department"`
	Enabled      bool           `db:"enabled"`
	DateCreated  time.Time      `db:"date_created"`
	DateUpdated  time.Time      `db:"date_updated"`
}
```

What is the problem with using `sql.Null`. We have the `to` functions to perform the data transformation between the layers.

**Comment by Dodje** [comment]
Yeah, I've found it in the `service` project `github.com/jackc/pgx/v5/`.
So I  made a use of `github.com/jackc/pgx/v5/pgtype` to work with geopoints `pgtype.Point`. That's why I've got an idea  to use `pgtypes` for postgres oriented structs..

> `What is the problem with using sql.NullString`

Actually that's kinda my mistake. I'm a newbie in go, so I now feel ashamed about not knowing about such simple things...)

Thank you for the quick and very useful repsonse!

**Comment by ardan-bkennedy** [comment]
No worries. I'm happy you asked the question and don't ever stop questioning things when they don't make sense.

I didn't notice they added this package to the module we already import. That is super confusing to have this package in two places. It wasn't necessary a bad idea, but I only use pgx for the driver. I am using sqlx to actually interact with the DB.
